.github/workflows/dependencies.yml

```diff
@@ -66,4 +66,23 @@ jobs:
       - name: Install cargo-machete
         run: cargo install cargo-machete --version ^0.9 --locked
       - name: Detect unused dependencies
-        run: cargo machete --with-metadata
\ No newline at end of file
+        run: cargo machete --with-metadata
+
+  dependency-graph:
+    name: dependency graph up to date
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8  # v6.0.1
+        with:
+          submodules: true
+          fetch-depth: 1
+      - name: Setup Rust toolchain
```

Comment:
I wonder if we should just run this script as part of some other job and save having to check  out the code again...